### PR TITLE
Add option to split the baseline grid.

### PR DESCRIPTION
**Pull request by loganj**
It's not uncommon for a design to call for bottom-anchored content, such
as buttons, snackbars, drawers, etc.  Such content should be
grid-aligned internally, but its grid should originate at the bottom of
the screen and may not align with the top-anchored grid.

This change adds an option to split the grid at a given point on the
screen. There will then be two grids, one originating at TOP and one at
BOTTOM. If the grids do not align, a "gutter" will be left between them
to indicate the odd space. The gutter will always be narrower than the
baseline grid size.

dspec_split_grid-2

### CODE REVIEW

**Comment by lucasr** [comment]
Or maybe even simpler: what about simply making the baseline grid 'split' point always be at the center of the drawable? That would cover TOP/BOTTOM/LEFT/RIGHT cases in most situations, no?

**Comment by loganj** [comment]
I'd originally intended to split in the center, but the designer I'm working with pointed out that we have content that extends beyond the midpoint but should be aligned to one grid or the other.  Right-aligned content seemed unusual enough not to bother with, but in retrospect that's pretty weak.

I like the gravity idea! Instead of a split grid, allow specifying a grid for each From origin, with optional max size.  Maybe specify color for each grid as well, allowing them to overlap.  Keep the existing methods for the basic behavior of a single unbounded grid anchored at TOP.

Something roughly like this:

```java
spec
  .setBaselineGrid(From.TOP, DesignSpec.UNBOUNDED, DesignSpec.DEFAULT_COLOR)
  .setBaselineGrid(From.BOTTOM, 48, res.getColor(android.R.color.white))
  // ...
```

If that sounds good to you, let's close this PR and I'll send you a new one.

**Comment by loganj** [comment]
The only ugly bit is that the limit's From needs to be parallel to the grid's From, but that can be checked at runtime.

**Comment by lucasr** [comment]
Hmm, the multi-baseline grid idea is looking a bit smelly to me to be honest i.e. a bit too complex, hard to follow. Unless I'm missing something here? You only want one baseline grid per design.

Stepping back a bit, I think we're better off with your original idea but let's expand it to account for horizontal split as well? Something like:
```json
"baselineGridVerticalSplit": {
  "offset": 48,
  "from": "BOTTOM"
}
"baselineGridHorizontalSplit": {
  "offset": 0,
  "from": "HORIZONTAL_CENTER"
}
```

I'm still mixed about the term "split" here because it doesn't quite match the semantics. It's more like a "grid convergence point" or something. Maybe ask the designer you're working with what term would make more sense for him/her? :-)

**Comment by lucasr** [comment]
Ah, and let's default to split at center of grid on both directions.
